### DISCUSSION TITLE
Kleine Verständnisfrage zu DID und DID Methoden

### DISCUSSION THREAD

**Discussion by mzhKU** [e-id-admin/open-source-community#20] [General]
Kann jemand erklären, was eine "did method" ist? Aus Preukschat et al. ist dieses Bild:

![image](https://user-images.githubusercontent.com/1569838/198218423-c4972835-83c8-4353-9cb2-5617c4c7cc93.png)

Die DID Methoden wären im Bild `sov`, `btcr`, ... Was kann man mit einer DID Methode machen, bzw. wie verwendet man sie? 

Eine andere Frage ist zu diesem Bild aus der gleichen Quelle:

![image](https://user-images.githubusercontent.com/1569838/198218749-7a25649f-45f0-407a-a99a-7461e8871052.png)

Hier wird illustriert, wie DID Resolution bei der Authentifizierung eingesetzt wird. Die Relying Party löst den DID Identifier des Login-Kandidaten auf, um einen öffentlichen Schlüssel der zum Login-Kandidaten gehört zu finden. Mit diesem Schlüssel wird ein "proof" verifiziert.

Was ist der Proof hier? Verschlüsselt die RP mit dem öffentlichen Schlüssel ein Token um zu prüfen ob der Login-Kandidat das Token entschlüsseln kann? Der Proof wäre dann das korrekt entschlüsselte Token, das der Kandidat zurück zur RP sendet um zu belegen, dass man im Besitz des Private Keys ist, der zum Public Key gehört.

**Comment by pascalgottret** [comment]
Sorry für die späte Antwort.

Du must dir vorstellen, dass sich das SSI Ecosystem nicht nur auf einer einzigen Datenbank (VDR, bzw. Basisregister) befindet. Die Schweiz möchte ggf. ein eigenen Register aufstellen, während EBSI sich für einen anderen entscheidet. Privat überlegst du dir evtl., dass du deine DIDs auf der Ethereum Blockchain oder auf dem IOTA DLT speicherst. Wir haben also eine grosse Auswahl.
Die DID Methode gibt einen Hinweis für den RP, auf Basis welcher Plattform eine DID aufgebaut ist. Diese Information braucht er softwaretechnisch, damit er weiss, wie die DID aufzulösen ist (DID Resolver). Mit dem Resolver kann er wichtige Daten aus deiner DID, besser gesagt aus dem öffentlichen DID Dokument, auslesen wie z.B. den Public Key und den Verschlüsselungsmechanismus.

Stell dir vor, zu zeigst deine eID einem Verifier. Die eID beinhaltet die DID vom Staat und dessen Signatur. Der Verifier kann nun mittels der DID Methode das DID Dokument "resolven", den Public Key auslesen und überprüfen, ob die Signatur deiner eID gültig ist.

Bezüglich der Authentifizierung ist der Proof nichts anderes als eine Signatur, mit dem bewiesen wird, dass man im Besitz der DID ist. Da eine DID self-certifying ist (steht ausführlich im Buch warum) kann die RP ohne Drittinstanz wie ein CA die Rechtmässigkeit überprüfen. Siehe dazu auch SIOPv2 https://openid.net/specs/openid-connect-self-issued-v2-1_0.html. Damit werden Passwörter für Logins der Vergangenheit gehören.

Falls du weitere Fragen hast, bitte einfach melden.
